main/inc/global.inc.php

```diff
@@ -80,6 +80,22 @@
 // Fix bug in IIS that doesn't fill the $_SERVER['REQUEST_URI'].
 api_request_uri();
 
+// Set web proxy environment variables
+foreach ([
+             api_get_configuration_sub_value('proxy_settings/stream_context_create/https/proxy'),
+             api_get_configuration_sub_value('proxy_settings/stream_context_create/http/proxy'),
+             api_get_configuration_sub_value('proxy_settings/curl_setopt_array/CURLOPT_PROXY'),
+         ] as $value) {
+    if ($value && is_string($value)) {
```

Comment:
I would prefer it if you used (!empty($value) && is_string($value)) here, because historically, PHP has not been very good at dealing with undefined variables properly (so a warning or notice could appear here depending on the PHP version and configuration.